tests/PathMappingRepositoryTest.php

```diff
@@ -131,6 +131,103 @@ public function testResolveFilesystemResource()
         $this->assertEquals(Path::normalize(__DIR__.'/Fixtures/dir5/sub'), $resource->getFilesystemPath());
     }
 
+    public function testMergeDirectParent()
```

Comment:
Can this test be moved to the abstract repository test? The same functionality should be provided by any repository implementation.